resolwe_bio/processes/go_enrichment/go.py

```diff
@@ -0,0 +1,143 @@
+        )
+        source = StringField(label="Gene ID database of selected genes")
+        species = StringField(
+            label="Species",
+            allow_custom_choice=True,
+            description="Specify species name. This field is required if gene subset is set.",
+            choices=[
+                ("Homo sapiens", "Homo sapiens"),
+                ("Mus musculus", "Mus musculus"),
+                ("Rattus norvegicus", "Rattus norvegicus"),
+                ("Dictyostelium discoideum", "Dictyostelium discoideum"),
+            ],
+        )
+        pval_threshold = FloatField(
+            label="P-value threshold", default=0.1, required=False
+        )
+        min_genes = IntegerField(
+            label="Minimum number of genes",
+            default=1,
+            required=False,
+            description="Minimum number of genes on a GO term.",
+        )
+
+    class Output:
+        """Output fields to process GOEnrichmentAnalysis."""
+
+        terms = JsonField(label="Enriched terms")
+        source = StringField(label="Source")
+        species = StringField(label="Species")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        if inputs.species != inputs.gaf.output.species:
+            self.warning(
+                "Selected genes Species must be the same as the Species field of the GAF file."
+            )
+            self.error(
+                f"Selected genes are from {inputs.species}, "
+                f"while GAF file has defined {inputs.gaf.output.species} under Species field."
+            )
+
+        org_features = self.feature.filter(
+            source=inputs.source, species=inputs.species, feature_id__in=inputs.genes
+        )
+
+        if len(org_features) == 0:
+            self.error("No genes were fetched from the knowledge base.")
+
+        if inputs.source == inputs.gaf.output.source:
+            target_ids = inputs.genes
+        else:
+            mapping_res = self.mapping.filter(
+                source_db=inputs.source,
+                source_species=inputs.species,
+                target_db=inputs.gaf.output.source,
+                target_species=inputs.gaf.output.species,
+                source_id__in=inputs.genes,
+            )
+
+            if len(mapping_res) == 0:
+                self.error("Failed to map features.")
+
+            mappings = {}
+            for m in mapping_res:
+                if m.source_id in inputs.genes:
+                    if m.source_id not in mappings:
+                        mappings[m.source_id] = m.target_id
+                    else:
+                        self.warning(f"Mapping {m} returned multiple times.")
+
+            if len(inputs.genes) > len(mappings):
+                self.warning("Not all features could be mapped.")
+
+            target_ids = mappings.values()
+
+        with tempfile.NamedTemporaryFile() as input_genes:
```

Comment:
This way of calling the `processor` tool from Python was necessary before the support for `plumbum` was introduced to Resolwe. I think you should be able to replace this part of the code with a standard `Cmd["processor"][...]()` call. If the tool emits to stdout, you can use something like `(Cmd["processor"][...] > "terms.json")()`